### DISCUSSION TITLE
Compile target for "esptool load_ram"

### DISCUSSION THREAD

**Discussion by TheChatty** [arendst/Tasmota#21368] [Ideas]
Hi,

it would be nice to have a compile target which produces a firmware image which can be run via "[esptool load_ram image.bin](https://docs.espressif.com/projects/esptool/en/latest/esp32s3/esptool/advanced-commands.html#load-a-binary-to-ram-load-ram)".

Excerpt:

> The load_ram command allows the loading of an executable binary image (created with the elf2image or make_image commands) directly into RAM, and then immediately executes the program contained within it.

**Comment by sfromis** [comment]
Does `load_ram` support loading the image into PSRAM? I did not see this suggested in the page you link to, and Tasmota images are way too large to fit into regular RAM of the ESP32.

**Comment by TheChatty** [comment]
`load_ram` does not support loading into PSRAM ootb. One would have to write a small proxy application for this.

I guess then this remains "a good idea".

**Comment by Jason2866** [comment]
There is not enough RAM to load Tasmota.
